### PR TITLE
Add `int64` out_idx` support for `listdiff`/`list_diff`/`setdiff1d`

### PR DESCRIPTION
**Pull request by yongtang**
This fix tries to add `int64` `out_idx` support for `listdiff`/`list_diff`/`setdiff1d`.

As was specified in docs (`tf.setdiff1d.__doc__`), it is possible to specify `tf.int32` or `tf.int64` for the type of the output idx. However, the `tf.int64` kernel has not been registered. As a consequence, an error will be thrown out if `tf.int64` is used.

This fix adds `int64` out_idx` support for `listdiff`/`list_diff`/`setdiff1d`. Related test cases have been updated as well.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Comment by vrv** [comment]
As usual, very nice!

@tensorflow-jenkins test this please

**Comment by vrv** [comment]
github issues it seems, trying again.  @tensorflow-jenkins test this please

File: tensorflow/python/kernel_tests/listdiff_op_test.py [hunk 1]
```diff
@@ -41,15 +41,17 @@ def _testListDiff(self, x, y, out, idx):
...
-        with self.test_session() as sess:
-          x_tensor = ops.convert_to_tensor(x, dtype=dtype)
-          y_tensor = ops.convert_to_tensor(y, dtype=dtype)
-          out_tensor, idx_tensor = diff_func(x_tensor, y_tensor)
-          tf_out, tf_idx = sess.run([out_tensor, idx_tensor])
-        self.assertAllEqual(tf_out, out)
-        self.assertAllEqual(tf_idx, idx)
-        self.assertEqual(1, out_tensor.get_shape().ndims)
-        self.assertEqual(1, idx_tensor.get_shape().ndims)
+        for index_dtype in [dtypes.int64]:
```
Comment (vrv) [inline tensorflow/python/kernel_tests/listdiff_op_test.py]: btw did you mean to test both int32 and int64?

Comment (yongtang) [reply tensorflow/python/kernel_tests/listdiff_op_test.py]: @vrv Forgot to add int32. Thanks a lot for pointing out!

**Comment by yongtang** [comment]
Thanks @vrv for the review. The PR has been updated. Please take a look.

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

**Review by vrv — APPROVED** [review]

**Pull request merged**